run_unittests.py

```diff
@@ -2994,6 +2996,14 @@ def test_vcs_tag_featurenew_build_always_stale(self):
         msg = '.*WARNING:.*feature.*build_always_stale.*custom_target.*'
         self.assertMesonDoesNotOutput(vcs_tag, msg, meson_version='>=0.43')
 
+    def test_feature_deprecation_messages(self):
+        self.assertMesonOutputs("import('python3')",
+                                ".*DEPRECATION.*Project targetting.*but.*deprecated.*",
+                                meson_version='>= 0.48.0')
+        self.assertMesonDoesNotOutput("import('python3')",
```

Comment:
I made that change because I didn't like how `assertMesonOutputs` was leaving behind a dirty builddir, so I made the change semantic on purpose. Of course I can change it the way you said if you think the semantic change is bad.

Reply:
Hm, then it is inconsistent with assertMesonRaises() above that would also leave a dirty builddir. I don't know if we have other cases. I personally think it's better to leave the builddir as-is because the test function could want to inspect what's inside.

btw, that would be self.wipe() instead of self.new_builddir() because the 2nd test doesn't need the builddir of the 1st test.

Anyway, that's all really bikeshedding, I'm fine either way ;-)

Reply:
> Hm, then it is inconsistent with assertMesonRaises() above that would also leave a dirty builddir

But that function can be run twice. i.e., you can run it twice in the same test and it'll be fine.

I'll add a `self.wipe()`.